File: services/mas-service/services/clients/python_client.js
import axios from 'axios';

const PYTHON_SERVICE_URL = process.env.PYTHON_SERVICE_URL || 'http://python-service:3003';

const client = axios.create({
    baseURL: PYTHON_SERVICE_URL,
    timeout: 600000, // 10 Minuten für Training
    headers: {
        'Content-Type': 'application/json'
    }
});

export const pythonClient = {
    // Data Analysis (mit LLM-Zusammenfassung)
    async analyzeData(filePath, forceRefresh = false) {
        try {
            const response = await client.post('/api/data/analyze', { filePath, forceRefresh });
            return response.data;
        } catch (error) {
            console.error('Fehler bei Data Analysis:', error.message);
            throw new Error(`Data Analysis fehlgeschlagen: ${error.response?.data?.error || error.message}`);
        }
    }
};

export default pythonClient;

File: services/unified-service/infrastructure/clients/python_client.py
"""
Python-Client für direkte Funktionsaufrufe
"""

from typing import Optional, Dict, Any
from core.data.data_exploration import get_cached_data_analysis

class PythonClient:
    """Client für direkte Funktionsaufrufe"""
    
    def analyze_data(self, file_path: str, force_refresh: bool = False) -> Dict[str, Any]:
        """Data Analysis (mit LLM-Zusammenfassung) - Direkter Funktionsaufruf"""
        try:
            result = get_cached_data_analysis(file_path, force_refresh)
            return result
        except Exception as error:
            print(f'Fehler bei Data Analysis: {error}')
            raise Exception(f'Data Analysis fehlgeschlagen: {error}')

# Globale Instanz
python_client = PythonClient()

